versioned/mongodb/src/main/java/com/dremio/nessie/versioned/impl/condition/BsonExpressionFunctionVisitor.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright (C) 2020 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.dremio.nessie.versioned.impl.condition;
+
+import java.security.InvalidParameterException;
+import java.util.List;
+
+import org.bson.conversions.Bson;
+
+import com.mongodb.client.model.Filters;
+
+/**
+ * This provides a separation of queries on @{ExpressionFunction} from the object itself.
+ * This uses the Visitor design pattern to retrieve object attributes.
+ */
+public class BsonExpressionFunctionVisitor implements ExpressionFunctionVisitor<Bson> {
+
+  /**
+   * This is a callback method that ExpressionFunction will call when this visitor is accepted.
+   * Its purpose is c reates a BSON representation of the ExpressionFunction object.
+   * @param expressionFunction the object to be converteds
+   * @return The ExpressionFunction represented as Bson
+   */
+  @Override
+  public Bson visit(ExpressionFunction expressionFunction, List<Value> arguments, ExpressionFunction.FunctionName name) {
```

Comment:
I would have preferred to expose these via getters but thought there might be push back. Using Visitor an an alternative to getters complicates the design. On the other hand I suppose it's obvious who is visiting. If the community would allow this then I would say it is a "nice to have". ie leave it until the end. Thoughts?

Reply:
I think we should make the change and then revert back if there's pushback.